### PR TITLE
Validate unique case names

### PR DESCRIPTION
**Pull request by seallard**
Validate case names are unique in order.

### CODE REVIEW

File: cg/services/order_validation_service/models/errors.py [hunk 1]
```diff
@@ -55,6 +55,11 @@ class OccupiedWellError(CaseSampleError):
     message: str = "Well is already occupied"
 
 
+class RepeatedCaseNameError(CaseError):
```
Comment (ChrOertlin) [inline cg/services/order_validation_service/models/errors.py:58]: nice

File: cg/services/order_validation_service/workflows/tomte/validation/inter_field/utils.py [hunk 2]
```diff
@@ -53,12 +54,23 @@ def _get_sample_well_map(plate_samples_with_cases: list[tuple[TomteSample, Tomte
     return sample_well_map
 
 
-def get_duplicate_sample_names(case: TomteCase) -> list[str]:
+def get_repeated_case_names(order: TomteOrder) -> list[str]:
+    case_names = [case.name for case in order.cases]
+    count = Counter(case_names)
+    return list({name for name, freq in count.items() if freq > 1})
```
Comment (ChrOertlin) [inline cg/services/order_validation_service/workflows/tomte/validation/inter_field/utils.py:60]:
```suggestion
    return [name for name, freq in count.items() if freq > 1]
```
Guess the same as with the sample?

Comment (ChrOertlin) [reply cg/services/order_validation_service/workflows/tomte/validation/inter_field/utils.py:60]: I wonder can that function be generalised so we only need 1 for both case and sample?
Even more, can you imagine this function to be required for other workflows as well?

```Python 
get_repeated_names(tome_model: TomteCase | TomteSample, associated_error: RepeatedCaseNameError | RepeatedSampleNameError)
```

**Review by ChrOertlin — APPROVED** [review]
Looks good.

File: cg/services/order_validation_service/workflows/tomte/validation/inter_field/utils.py (hunk 2 above)
Comment (seallard) [reply cg/services/order_validation_service/workflows/tomte/validation/inter_field/utils.py:60]: I'll keep them separate for now, that signature looks confusing! Will consider it in the future if there is a lot of duplication

Comment (ChrOertlin) [reply cg/services/order_validation_service/workflows/tomte/validation/inter_field/utils.py]: The signature was just an example to make clearer what I meant. and sounds good!

**Pull request merged**